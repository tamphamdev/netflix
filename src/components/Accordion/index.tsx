import React, {
  useState,
  useContext,
  createContext,
  Dispatch,
  SetStateAction,
} from "react"
import {
  Container,
  Frame,
  Title,
  Item,
  Inner,
  Header,
  Body,
} from "../Accordion/styles/accordion"
type Props = {
  children: React.ReactNode
}

type ContextType = {
  toggleShow: boolean
  setToggleShow: (value: boolean) => void
}

const contextDefaultValues: ContextType = {
  toggleShow: false,
  setToggleShow: (): void => {},
}

const ToggleContext = createContext(contextDefaultValues)
export default function Accordion({ children, ...restProps }: Props) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  )
}

Accordion.Title = function AccordionTitle({ children, ...restProps }: Props) {
  return <Title {...restProps}>{children}</Title>
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }: Props) {
  return <Frame {...restProps}>{children}</Frame>
}

Accordion.Item = function AccordionItem({ children, ...restProps }: Props) {
  const [toggleShow, setToggleShow] = useState(false)

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  )
}

Accordion.Header = function AccordionHeader({ children, ...restProps }: Props) {
  const { toggleShow, setToggleShow } = useContext(ToggleContext)

  return (
    <Header
      onClick={() => setToggleShow((toggleShow) => !toggleShow)}
      {...restProps}
    >
      {children}
      <pre>{JSON.stringify(toggleShow, null, 2)}</pre>
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  )
}

Accordion.Body = function AccordionBody({ children, ...restProps }: Props) {
  const { toggleShow } = useContext(ToggleContext)

  return toggleShow ? <Body {...restProps}>{children}</Body> : null
}
