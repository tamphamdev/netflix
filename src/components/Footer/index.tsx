import React from "react";
import {
  Container,
  Row,
  Column,
  Link,
  Text,
  Title,
  Break,
} from "./styles/footer";

type Props = {
  children?: React.ReactNode;
};

export default function Footer({ children, ...restProps }: Props) {
  return <Container {...restProps}>{children}</Container>;
}

Footer.Row = function FooterRow({ children, ...restProps }: Props) {
  return <Row {...restProps}>{children}</Row>;
};

Footer.Column = function FooterColumn({ children, ...restProps }: Props) {
  return <Column {...restProps}>{children}</Column>;
};

Footer.Link = function FooterLink({ children, ...restProps }: Props) {
  return <Link {...restProps}>{children}</Link>;
};

Footer.Title = function FooterTitle({ children, ...restProps }: Props) {
  return <Title {...restProps}>{children}</Title>;
};

Footer.Text = function FooterText({ children, ...restProps }: Props) {
  return <Text {...restProps}>{children}</Text>;
};

Footer.Break = function FooterBreak({ children, ...restProps }: Props) {
  return <Break {...restProps}>{children}</Break>;
};
