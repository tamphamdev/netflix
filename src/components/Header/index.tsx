import React, { Children } from "react"
import { Link as ReactRouterLink } from "react-router-dom"
import {
  Background,
  Container,
  Logo,
  Group,
  Picture,
  ButtonLink,
  Feature,
  Dropdown,
  Text,
  Profile,
  FeatureCallOut,
  Link,
} from "./styles/header"

type Props = {
  to?: string
  background?: boolean
  children: any
  [x: string]: any
}
type PictureProps = {
  src: string
  [x: string]: any
}

type LogoProps = {
  to: string
  [x: string]: any
}
export default function Header({
  background = true,
  children,
  ...restProps
}: Props) {
  return background ? (
    <Background {...restProps}>{children}</Background>
  ) : (
    children
  )
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Picture = function HeaderPicture({ src, ...restProps }: PictureProps) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }: Props) {
  return <Link {...restProps}>{children}</Link>
}

Header.Frame = function HeaderFrame({ children, ...restProps }: Props) {
  return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }: Props) {
  return <Group {...restProps}>{children}</Group>
}

Header.Logo = function HeaderLogo({ to, ...restProps }: LogoProps) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.ButtonLink = function HeaderButtonLink({
  children,
  ...restProps
}: Props) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}
