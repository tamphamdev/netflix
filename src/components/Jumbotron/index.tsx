import React from "react";
import {
  Container,
  Item,
  Inner,
  Pane,
  Title,
  SubTitle,
  Image,
} from "./styles/jumbotron";

type Props = {
  direction: string;
  children: React.ReactNode;
};
export default function Jumbotron({
  children,
  direction = "row",
  ...restProps
}: Props) {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
}

Jumbotron.Container = function JumbotronContainer({
  children,
  ...restProps
}: Props) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }: Props) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }: Props) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumbotronSubTitle({
  children,
  ...restProps
}: Props) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }: Props) {
  return <Image {...restProps} />;
};
