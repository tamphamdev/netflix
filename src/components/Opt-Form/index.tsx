import React from "react"
import { Container, Input, Button, Text, Break } from "./styles/opt-form"

type Props = {
  children: React.ReactNode
}

export default function OptForm({ children, ...resProps }: Props) {
  return <Container {...resProps}>{children}</Container>
}

OptForm.Input = function OptFormInput({ ...restProps }) {
  return <Input {...restProps} />
}

OptForm.Button = function OptFormButton({ children, ...restProps }: Props) {
  return (
    <Button {...restProps}>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try Now" />
    </Button>
  )
}

OptForm.Text = function OptFormText({ children, ...restProps }: Props) {
  return <Text {...restProps}>{children}</Text>
}

OptForm.Break = function OptFormBreak({ ...restProps }) {
  return <Break {...restProps} />
}
