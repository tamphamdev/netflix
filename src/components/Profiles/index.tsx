import React from "react"
import {
  Container,
  List,
  Item,
  Picture,
  Name,
  Title,
  User,
} from "./styles/profiles"
type Props = {
  children?: React.ReactChild
  [x: string]: any
  src?: string
}
export default function Profiles({ children, ...restProps }: Props) {
  return <Container {...restProps}>{children}</Container>
}

Profiles.Title = function ProfilesTitle({ children, ...restProps }: Props) {
  return <Title {...restProps}>{children}</Title>
}

Profiles.Item = function ProfilesItem({ children, ...restProps }: Props) {
  return <Item {...restProps}>{children}</Item>
}

Profiles.Picture = function ProfilesPicture({ src, ...restProps }: Props) {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`}
    />
  )
}

Profiles.Name = function ProfilesName({ children, ...restProps }: Props) {
  return <Name {...restProps}>{children}</Name>
}

Profiles.List = function ProfilesList({ children, ...restProps }: Props) {
  return <List {...restProps}>{children}</List>
}

Profiles.User = function ProfilesUser({ children, ...restProps }: Props) {
  return <Item {...restProps}>{children}</Item>
}
