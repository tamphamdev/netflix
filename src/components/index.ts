export { default as Accordion } from "./Accordion"
export { default as Footer } from "./Footer"
export { default as Jumbotron } from "./Jumbotron"
