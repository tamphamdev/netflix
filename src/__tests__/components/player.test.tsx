import { fireEvent, render } from "@testing-library/react"
import React from "react"
import { Player } from "../../components"

describe("<Player/>", () => {
  it("render the <Player/> with a buny video", () => {
    const { getByText, queryByTestId, container, debug } = render(
      <Player>
        <Player.Button />
        <Player.Video src="/public/videos/bunny.mp4" />
      </Player>
    )
    debug()
    expect(queryByTestId("player")).toBeFalsy()
    fireEvent.click(getByText("Play"))

    expect(queryByTestId("player")).toBeTruthy()
    fireEvent.click(queryByTestId("player"))

    expect(queryByTestId("player")).toBeFalsy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
