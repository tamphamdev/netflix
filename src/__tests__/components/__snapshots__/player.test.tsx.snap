// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player/> render the <Player/> with a buny video 1`] = `
<div
  class="sc-dvQaRk"
>
  <button
    class="sc-jIkXHa glrNqN"
  >
    Play
  </button>
</div>
`;
